src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +78,279 @@ public class AnalysisService : IDisposable
+        /// <param name="forBlockComment">If true, block comment help will be supplied.</param>
+        /// <returns></returns>
+        public async Task<string> GetCommentHelpText(string functionText, string helpLocation, bool forBlockComment)
         {
-            // We cannot use Range type therefore this workaround of using -1 default value.
-            // Invoke-Formatter throws a ParameterBinderValidationException if the ScriptDefinition is an empty string.
-            if (string.IsNullOrEmpty(scriptDefinition))
+            if (AnalysisEngine == null)
             {
                 return null;
             }
 
-            var argsDict = new Dictionary<string, object> {
-                    {"ScriptDefinition", scriptDefinition},
-                    {"Settings", settings}
-            };
-            if (rangeList != null)
-            {
-                argsDict.Add("Range", rangeList);
-            }
+            Hashtable commentHelpSettings = AnalysisService.GetCommentHelpRuleSettings(helpLocation, forBlockComment);
 
-            PowerShellResult result = await InvokePowerShellAsync("Invoke-Formatter", argsDict).ConfigureAwait(false);
+            ScriptFileMarker[] analysisResults = await AnalysisEngine.AnalyzeScriptAsync(functionText, commentHelpSettings).ConfigureAwait(false);
 
-            if (result == null)
+            if (analysisResults.Length == 0
+                || analysisResults[0]?.Correction?.Edits == null
+                || analysisResults[0].Correction.Edits.Count() == 0)
             {
-                _logger.LogError("Formatter returned null result");
                 return null;
             }
 
-            if (result.HasErrors)
-            {
-                var errorBuilder = new StringBuilder().Append(s_indentJoin);
-                foreach (ErrorRecord err in result.Errors)
-                {
-                    errorBuilder.Append(err).Append(s_indentJoin);
-                }
-                _logger.LogWarning($"Errors found while formatting file: {errorBuilder}");
-                return null;
-            }
+            return analysisResults[0].Correction.Edits[0].Text;
+        }
 
-            foreach (PSObject resultObj in result.Output)
+        /// <summary>
+        /// Get the most recent corrections computed for a given script file.
+        /// </summary>
+        /// <param name="documentUri">The URI string of the file to get code actions for.</param>
+        /// <returns>A threadsafe readonly dictionary of the code actions of the particular file.</returns>
+        public async Task<IReadOnlyDictionary<string, MarkerCorrection>> GetMostRecentCodeActionsForFileAsync(string documentUri)
+        {
+            // On Windows, VSCode still gives us file URIs like "file:///c%3a/...", so we need to escape them
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
is this really only on Windows? jeez...

Also, you should pass in a `Uri` instead and use `WorkspaceService.ResolveFileUri(Uri fileUri)`.

Please test this out with non-ASCII characters.